cmd/deploy/deploy.go

```diff
@@ -573,15 +576,15 @@ func getDefaultTimeout() time.Duration {
 }
 
 func shouldRunInRemote(opts *Options) bool {
-	// --remote flag enabled from command line
-	if opts.RunInRemote {
-		return true
+	// --remote flag enabled from command line, either true or false the flag value takes precedence
+	if opts.RunInRemoteSet {
+		return opts.RunInRemote
 	}
 
-	// remote option set in the manifest via a remote deployer image or the remote option enabled
+	// remote option set in the manifest via the remote option enabled
 	if opts.Manifest != nil && opts.Manifest.Deploy != nil {
-		if opts.Manifest.Deploy.Image != "" || opts.Manifest.Deploy.Remote {
-			return true
+		if opts.Manifest.Deploy.Remote != nil {
```

Comment:
with this change, if I have `image:` defined, it doesn't necessarily remote deploy, correct?

If that's the case, it sounds a breaking change somehow, no? I couldn't find reference to it in the specs 🤔

Reply:
This is part of the migration path, there is a block in the spec for this. i might have missunderstood smthg. i am reviewing this. thanks!

Reply:
We need to double check this, because there was a comment from Pablo in the spec a bit confusing

Reply:
added a comment on the spec to get some answers to this. thanks

Reply:
Updated PR and open for reviews